my-house/src/components/TODOList/TODOList.tsx

```diff
@@ -1,32 +1,85 @@
-"use client"
-
-import styles from "./TODOList.module.scss";
+import { TodosType } from '@/app/TODO-list/page'
+import TODOItem from './TODOItem'
+import styles from './TODOList.module.scss'
 import React from 'react'
+import TODOAddItem from './TODOAddItem'
+import { v4 as uuidv4 } from 'uuid'
 
-interface Task {
-  order: number,
-  id: string,
-  text: string
+export interface Task {
+    order: number
+    text: string
+    checked: boolean
+    id: string
 }
-interface Props {
-  name: string,
-  order: number,
-  tasks: Array<Task>,
-  id: string
+export interface Props {
+    onChange: (todo: TodosType) => void
+    TODO: TodosType
+    key: string
 }
 
-const TODOList = (props: Props) => {
-  return (
-    <section className={styles.TODOList}>
-        <header className={styles.TODOList__header}>
-            <h3 className={styles.TODOList__name}>{props.name}</h3>
-            <div className={styles.TODOList__settingsButton}>...</div>
-        </header>
-        <ul className={styles.TODOList__tasks}>
-        {props.tasks?.map((task) => <li id={task.id} className={styles.TODOList__task}>{task.text}</li>)}
-        </ul>
-    </section>
-  )
+const TODOList = ({ onChange, TODO }: Props) => {
+    const { name, tasks } = TODO
+    const handleTaskUpdate = (updatedTask: Task) => {
+        const updatedTasks = tasks.map((task) => {
+            if (task.id === updatedTask.id) {
+                return updatedTask
+            }
+            return task
+        })
+        onChange({ ...TODO, tasks: updatedTasks })
+    }
+    const handleTaskDelete = (taskId: string) => {
+        const updatedTasks = tasks.filter((task) => task.id !== taskId)
+        onChange({ ...TODO, tasks: updatedTasks })
+    }
+    const handleTaskAdd = (text: string) => {
+        const newOrder = tasks.sort((a, b) => a.order - b.order)[0].order + 1
```

Comment:
```suggestion
        const newOrder = tasks.length + 1
```
raczej nie zakładamy że będziemy mieć orderów w stylu `[1,2,8]`, ale jeśli taki przypadek może być i jest wymagany, to wtedy twoje rozwiązanie jest ok, chodź te ordery przy tworzeniu B/E może ustawiać